src/api/WebLink.js

```diff
@@ -28,6 +29,70 @@ class WebLink extends Item {
         const suffix: string = id ? `/${id}` : '';
         return `${this.getBaseApiUrl()}/web_links${suffix}`;
     }
+
+    /**
+     * Gets a Box weblink
+     *
+     * @param {string} id - Weblink id
+     * @param {(newItem: BoxItem) => void} successCallback - Function to call with results
+     * @param {Function} errorCallback - Function to call with errors
+     * @param {Array<String>} fields - Array of field strings
+     * @returns {Promise}
+     */
+    async getWeblink(
+        id: string,
+        successCallback: (newItem: BoxItem) => void,
+        errorCallback: Function,
+        { fields }: FetchOptions = {},
+    ): Promise<void> {
+        if (this.isDestroyed()) {
+            return;
+        }
+
+        const cache: APICache = this.getCache();
+        const key: string = this.getCacheKey(id);
+        this.errorCode = ERROR_CODE_FETCH_WEBLINK;
+        this.successCallback = successCallback;
+        this.errorCallback = errorCallback;
+
+        if (cache.has(key)) {
+            const missingFields: Array<string> = findMissingProperties(cache.get(key), fields);
+            if (missingFields.length === 0) {
+                successCallback(cache.get(key));
+                return;
+            }
+        }
+
+        const xhrOptions: Object = {
+            url: this.getUrl(id),
+        };
+
+        if (fields) {
+            xhrOptions.params = {
+                fields: fields.toString(),
+            };
+        }
+
+        try {
+            const { data } = await this.xhr.get(xhrOptions);
+
+            if (this.isDestroyed()) {
+                return;
+            }
+
+            // Cache check is again done since this code is executed async
+            if (cache.has(key)) {
```

Comment:
This code looks familiar. I'm wondering if we can update `cache.merge` to support unset keys to avoid this check. @priyajeet, do you have a preference?

Reply:
Copied from File.js  😂